New functions for checking multiple sets

linuswagner opened the pull request:
Add two functions to `Set` library:
1. `intersection` to get the intersection of multiple sets
2. `isDisjoined` to check if multiple sets are pairwise disjoined

jurgenvinju commented on src/org/rascalmpl/library/Set.rsc:
```diff
@@ -513,3 +513,29 @@ public set[&T] union(set[set[&T]] sets) = {*s | s <- sets};
+
+
+@synopsis{Calculate the intersection of a set of sets.}
+public set[&T] intersection(set[set[&T]] sets)  = (getFirstFrom(sets) | it & elem | elem <- sets);
```
You could write this:
```
public set[&T] intersection({set[&T] firstSet, *set[&T] otherSets}) = (firstSet | it & elem | elem <- otherSets);
```

However, the function would then still throw an exception on the empty set (like the current version). If you want to avoid that add this alternative overload:

```
public set[&T] intersection({}) = {};
```

jurgenvinju commented on src/org/rascalmpl/library/Set.rsc:
````diff
@@ -513,3 +513,29 @@ public set[&T] union(set[set[&T]] sets) = {*s | s <- sets};
+isDisjoined({{1,2}, {1,4}, {1,6}});
+```
+}
+public bool isDisjoined(set[set[&T]] sets) {
````
spelling error: should be "disjoint"

jurgenvinju commented on src/org/rascalmpl/library/Set.rsc:
```diff
@@ -513,3 +513,29 @@ public set[&T] union(set[set[&T]] sets) = {*s | s <- sets};
+
+  for (elem1 <- [0..size(setsAsList)-1]) {
+    for (elem2 <- [elem1+1..size(setsAsList)]) {
+      if (setsAsList[elem1] & setsAsList[elem2] != {}) return false;
```
pls add a newline and brackets?

jurgenvinju replied:
You could avoid the code for generating all combinations by using set matching. It will produce the exact same combinations, and we still use your smart trick of reusing the fast intersection operator:
 
```
bool isDisjoint({set[&T] a, set[&T] b, *_}) = false when a & b != {}; // will backtrack to other pairs of a and b while the condition fails
default bool isDisjoint({set[&T] a, set[&T] b, *_}) = true;
```

jurgenvinju commented:
Very nice additions and necessary to the library. Thanks @linuswagner

DavyLandman commented:
> > Hi @linuswagner,
> > thanks for the nice addition. Could you at at least a few tests to the [lang::rascal::tests::library::Set](https://github.com/usethesource/rascal/blob/main/src/org/rascalmpl/library/lang/rascal/tests/library/Set.rsc) module?
> 
> Perfect, that's what I was looking for. Thanks!

Cool 👍🏼 

> 
> I've decided to make the operations stricter and throw exceptions when the input has not at least 2 elements. The decision here is that I rather have something that is supposed to compare tell me if it can't compare rather than that it masks that. On the downside, others will always have to check before calling the function, which causes more code and could cause weird cornercases (e.g. most of the times, the program produces inputs with at least 2 sets, but rarely, it gives less than that).
> 
> Is Rascal generally "better safe than sorry" or "you should better know what you're doing"?

I think better safe than sorry. We don't want someone to base their conclusion on something where rascal was just sneakily hiding an error the user made. So if there is no default correct behavior for an input with 1 element, then yeah, throw an exception.

DavyLandman commented on src/org/rascalmpl/library/Set.rsc:
```diff
@@ -513,3 +513,53 @@ public set[&T] union(set[set[&T]] sets) = {*s | s <- sets};
+public bool isPairwiseDisjoint(wholeInput:list[set[&T]] sets) {
+  int sizeSets = size(sets);
+  if (sizeSets == 0 || sizeSets == 1) {
+    throw IllegalArgument(wholeInput, "Only two or more sets can be pairwise disjoint.");
```
`sets` == `wholeInput`, no need to give it two names right?

DavyLandman commented on src/org/rascalmpl/library/lang/rascal/tests/library/Set.rsc:
```diff
@@ -157,6 +157,51 @@ test bool testDynamicTypes2() {set[value] s = {"1",2,3}; return set[int] _ := s
+
+test bool testIsPairwiseDisjointIdenticalElements() {return isPairwiseDisjoint([{1}, {1}]) == false;}
+test bool testIsPairwiseDisjointNoOverlap() {return isPairwiseDisjoint([{1,2},{3,4},{5,6}]) == true;}
+test bool testIsPairwiseDisjointOverlap() {return isPairwiseDisjoint([{1,2}, {-4,5}, {1,6,7}]) == false;}
```
if possible, think of an way to write an random tests. something like:

```rascal
test bool testRandomPairwiseDisjoint(set[value] a, set[value] b) = (a & b == {}) == isPairwiseDisjoint([a,b]);
```

or even better if you could do it without using the intersection operator.

PaulKlint committed:
Made messages related to ignore more precise.

toinehartman committed:
Ignore test until we find a robust approach.

jurgenvinju committed:
cleaning up parameters and error reporting

jurgenvinju committed:
fixed broken header

PaulKlint committed:
Merge branch 'main' into fix/integration-test-errors2

DavyLandman committed:
Merge pull request #2646 from usethesource/fix/limit-error-range-ignores

Made messages related to ignore more precise.

DavyLandman committed:
Merge pull request #2636 from usethesource/fix/integration-test-errors2

Fix errors in integration tests 2

jurgenvinju committed:
testing.json is now longer to no longer need the carriage returns to trigger #2633

toinehartman committed:
Fix doc tags, spelling errors and typos in library docs.

toinehartman committed:
Fix more doc tags, spelling errors and typos.

jurgenvinju committed:
testing and working on asserts

DavyLandman committed:
Merge pull request #2647 from usethesource/docs/fix-library-spelling

Fix doc tags, spelling errors and typos in library documentation

toinehartman committed:
Fix Prelude documentation (#2639)

* Add missing modules in Prelude documentation.

* Remove trailing spaces.

PaulKlint committed:
Check body of ignored test functions

PaulKlint committed:
Added missing import (unrelated to PR)

PaulKlint committed:
Uopgraded to typepal 0.16.4

jurgenvinju committed:
make sure tracking is really of when a JsonReader is passed in, to avoid NPEs and other crashes

jurgenvinju committed:
forgot to implement the degenerate parse error

jurgenvinju committed:
cleaning up

jurgenvinju committed:
added some comments

jurgenvinju committed:
reset testing.json

jurgenvinju committed:
commented read better and added fix by @DavyLandman for when off!=0

DavyLandman committed:
Merge pull request #2648 from usethesource/fix/check-ignored-test

Fix/check ignored test

jurgenvinju committed:
added tests by @davylandman and fixed off-by-one with length of origin positions on objects

jurgenvinju committed:
fixing off-by-ones

jurgenvinju committed:
reset

toinehartman committed:
Remove all layout from definition signatures.

toinehartman committed:
Align with Unicode standard instead of Rascal layout spec.

toinehartman committed:
Fix typo in tutor documentation.

toinehartman committed:
Merge remote-tracking branch 'origin/main' into fix/os-dependent-logical-locs

toinehartman committed:
Increase major version to better represent extent of changes.

toinehartman committed:
Simplify whitespace check using Unicode categories.

toinehartman committed:
Rewrite: readability.

PaulKlint committed:
Added missing import

PaulKlint committed:
New test scenarios with outdated TPL versions

PaulKlint committed:
Better handling of rascalTplVersionError

PaulKlint committed:
Better handling of rascalTplVersionError

PaulKlint committed:
Improved message

PaulKlint committed:
Fixed type errors in ignored tests

PaulKlint committed:
Renamed ignoreCompiler -> hasIgnoreCompilerTag; Adapted definition

PaulKlint committed:
Fixed renaming issue

DavyLandman committed:
Merge pull request #2653 from usethesource/fix/better-handling-of-ignore

Fix/better handling of ignore

PaulKlint committed:
Set verbose to false

PaulKlint committed:
Merge branch 'main' into feat/new-version-tests

DavyLandman committed:
Merge pull request #2652 from usethesource/feat/new-version-tests

New test scenarios for handling TPLs with outdated version number

toinehartman committed:
Separate individual hash contributions.

toinehartman committed:
Remove unused hash contribution rules.

toinehartman committed:
Merge remote-tracking branch 'origin/main' into fix/os-dependent-logical-locs

PaulKlint committed:
Merge pull request #2644 from usethesource/fix/os-dependent-logical-locs

OS independent logical locs

PaulKlint committed:
Moved declaration of normalizedMD5Hash to BasicRascalConfig

PaulKlint committed:
Replaced two nested parses (in error situation) by a dummy parse tree.

jurgenvinju committed:
refactored and simplified use of the wrapped reader. Also improved tests to include line/column accuracy testing. Fixed some off-by-one errors

jurgenvinju committed:
removed unused readCount

jurgenvinju committed:
ignored a test again

PaulKlint committed:
Fixed type error

jurgenvinju committed:
Merge branch 'main' into fix/json-offsets

PaulKlint committed:
Fixed another type error

jurgenvinju committed:
removed test printlns

PaulKlint committed:
Removed two unused files that cause type errors

PaulKlint committed:
Fixed small errors found during integration testing

PaulKlint committed:
Merge pull request #2655 from usethesource/fix/two-issues-in-integration-tests

Fix/two-issues-in-integration-tests

PaulKlint committed:
Merge branch 'main' into fix/more-integration-tests

PaulKlint committed:
Merge branch 'main' into fix/remove-two-unused-files

PaulKlint committed:
Merge pull request #2656 from usethesource/fix/remove-two-unused-files

Removed two unused files that cause type errors

PaulKlint committed:
Merge branch 'main' into cleanup-atree

PaulKlint committed:
Merge pull request #2536 from usethesource/cleanup-atree

removed unused definition of ATree to avoid confusion

PaulKlint committed:
Merge branch 'main' into replace-char-by-char-class-atype

toinehartman committed:
Revert "if watchers detected a module has been removed, we should not try to reload it" (#2657)

This reverts commit ef973c80e99fd481b703015f1d717b7a176a7714 from #2595,
since it breaks module reloading (#2654).

Co-authored-by: Jurgen J. Vinju <Jurgen.Vinju@cwi.nl>

PaulKlint committed:
Merge branch 'main' into replace-char-by-char-class-atype

PaulKlint committed:
Merge pull request #2533 from usethesource/replace-char-by-char-class-atype

removed ununsed asubtype relation between a ATree (achar) and a AType (\achar-class) to avoid confusion between types and values.

PaulKlint committed:
Merge branch 'main' into fix/more-integration-tests

PaulKlint committed:
Merge pull request #2658 from usethesource/fix/more-integration-tests

Fixed small errors found during integration testing

DavyLandman committed:
[maven-release-plugin] prepare release v0.42.0-RC1-BOOT

DavyLandman committed:
[maven-release-plugin] prepare for next development iteration

DavyLandman committed:
Using boot release

DavyLandman committed:
[maven-release-plugin] prepare release v0.42.0-RC1

DavyLandman committed:
[maven-release-plugin] prepare for next development iteration

DavyLandman committed:
Using rc for rascal bootstrap

[skip ci]

jurgenvinju committed:
removed dead import

jurgenvinju committed:
Merge branch 'main' into fix/json-offsets

jurgenvinju committed:
fixed comment by @davylandman

jurgenvinju committed:
Merge pull request #2643 from usethesource/fix/json-offsets

Fixing Json character offsets via wrapping the underlying character buffer"

jurgenvinju committed:
renamed Environment.getLocation to Environment.getCreatorLocation for clarity. Used heap.getModuleURI to check if a module file still exists or not (#2660)

PaulKlint committed:
Added import that is used from ignored test

PaulKlint committed:
Commented out test that is problematic for evaluator and compiler

PaulKlint committed:
Merge pull request #2661 from usethesource/fix/ignored-tests

Fix/ignored-tests

PaulKlint committed:
Removed no longer relevant files

PaulKlint committed:
Merge branch 'main' into fix/remove-two-unused-files

PaulKlint committed:
Merge pull request #2663 from usethesource/fix/remove-two-unused-files

Removed no longer relevant files

PaulKlint committed:
Fixed typo

PaulKlint committed:
Replaced Ignore -> IgnoreCompiler (#2664)

* Replaced Ignore -> IgnoreCompiler

* "Double ignore" test

* "Double ignore" test

PaulKlint committed:
Fix/effect-ignore-in-tests (#2665)

* Added missing import

* Fixed missing return

PaulKlint committed:
Fix/packager-without-guard-rails (#2662)

* Removed safe guard against malfunctioning packager

* Removed check on logical locs that seemed necessary

* Fixed error

PaulKlint committed:
Improved message for outdated TPL and removed possibly confusing println

PaulKlint committed:
Reorganized the reporting of TPL version errors

PaulKlint committed:
Removed println

PaulKlint committed:
Fixed quotes

PaulKlint committed:
Merge branch 'main' into fix/outdated-tpl-message

DavyLandman committed:
Merge pull request #2666 from usethesource/fix/outdated-tpl-message

Improved message for outdated TPL and removed possibly confusing println

DavyLandman committed:
[maven-release-plugin] prepare release v0.42.0-RC2

DavyLandman committed:
[maven-release-plugin] prepare for next development iteration

toinehartman committed:
Fix Maven not resolving parent dependencies (#2672)

* Add failing test for #2668.

* Fix documentation typos.

* Fix parent of sibling resolver.

* Update Maven deps to 3.9.12.

jurgenvinju committed:
fix/issue 2679 (#2680)

* make sure errors are attached with the importing module if the imported module does not exist anymore to prevent later NPEs and loss of information on the current state of the import graph

* changed reloader root location from reloader:/// to prompt:/// to remove a superfluous scheme and to highlight to the user that the origin of the reloading mechanism is the REPL. If a module was imported by the REPL and the reload fails because the file has dissappeared, then this will be the error location

PaulKlint committed:
Fix undesired propagation of type errors from TModels to ModuleStatus

PaulKlint committed:
Fixed layout

PaulKlint committed:
Added test for disappering error messages ht @toinehartman

PaulKlint committed:
Fixed writeModules

PaulKlint committed:
Provide rascal version for new test

PaulKlint committed:
Avoid the "Cannot add grammar ..." message (#2677)

PaulKlint committed:
Upgraded to typepal 0.16.6-RC1 and fixes for useViaType (#2667)

* Upgraded to typepal 0.16.5

* Updated to typepal 0.16.6-RC1

* Added test

* Reverted recent change

DavyLandman committed:
Merge pull request #2681 from usethesource/fix/persisting-type-errors

Fix/persisting-type-errors

DavyLandman committed:
[maven-release-plugin] prepare release v0.42.0-RC3

DavyLandman committed:
[maven-release-plugin] prepare for next development iteration

jurgenvinju committed:
renamed cause keyword parameter to "reason" to comply with the internal definition in RuntimeExceptionFactory

This also resolves a duplicate declaration error with Exception.rsc where `cause` is of type `RuntimeException` and not `str` for the `Java` and `JavaException` constructors.

jurgenvinju committed:
Merge pull request #2682 from usethesource/fix/json-cause-field

renamed cause keyword parameter to "reason" to comply with the interal definition in RuntimeExceptionFactory

jurgenvinju committed:
work on #2683 (#2685)

* fixes #2683 but reveals another hidden bug behind it. the REPL module is not re-extending its extended modules when the watcher has seen a modification upstream

* shell environment can not extend modules anymore. changes to import automatically with a warning

PaulKlint committed:
Feat/new-destructuring-test (#2686)

* New test combinbing destructuring assignment and field access ht @toinehartman

* Upgraded to typepal 0.16.6-RC2

PaulKlint committed:
Cleanup triggered by integration tests (#2687)

DavyLandman committed:
[maven-release-plugin] prepare release v0.42.0

DavyLandman committed:
[maven-release-plugin] prepare for next development iteration

jurgenvinju committed:
minor issue in new includeLibraries feature; IO fail if the library dependency is not a Rascal project with a docs folder

jurgenvinju committed:
fixes #2691 and fixes #2689

jurgenvinju committed:
cleanup

PaulKlint committed:
Removed too agressive check

PaulKlint committed:
Streamlined code

PaulKlint committed:
Added tests for parameterized types

PaulKlint committed:
Merge branch 'main' into fix/type-parameter-checking

PaulKlint committed:
Refined check on type parameters

PaulKlint committed:
Added tests

PaulKlint committed:
Added test

PaulKlint committed:
Adapted test since different error is generated.

DavyLandman committed:
Merge pull request #2693 from usethesource/fix/type-parameter-checking

Fix/type-parameter-checking

jurgenvinju committed:
programmatic uses of doImport forgot to clear lookup caches afterwards.  (#2695)

Fixes #2694

jurgenvinju committed:
Merge branch 'main' into fix/issues-2691-2689-2690

DavyLandman committed:
Using latest release for bootstrap

DavyLandman committed:
[maven-release-plugin] prepare release v0.42.1

DavyLandman committed:
[maven-release-plugin] prepare for next development iteration

jurgenvinju committed:
fixed a bunch of corner cases around code blocks and comments inside of them

jurgenvinju committed:
fixed TODO markup

jurgenvinju committed:
fixed more corner cases by using recursion instead of iteration for scraping off empty lines and comment lines in shell blocks

jurgenvinju committed:
rascal-commands are now also split over commented lines

PaulKlint committed:
Added missing import

PaulKlint committed:
Merge branch 'main' into fix/missing-import2

jurgenvinju committed:
forgot closing quotes

jurgenvinju committed:
Merge branch 'main' into fix/issues-2691-2689-2690

PaulKlint committed:
Removed unwanted escaping for Java keywords in file paths

PaulKlint committed:
Replaced asBaseClassName by asFileName to avoid escaping of Java keywords

PaulKlint committed:
Fixed casualty of renaming

PaulKlint committed:
Updated list of Java keywords (used for escaping names in generated code)

PaulKlint committed:
Merge pull request #2706 from usethesource/fix/escape-in-path

Fix/escape in path

PaulKlint committed:
Merge pull request #2704 from usethesource/fix/missing-import2

Added missing import

PaulKlint committed:
Merge branch 'main' into fix/upgrade-java-reserved-keywords

PaulKlint committed:
Merge pull request #2707 from usethesource/fix/upgrade-java-reserved-keywords

Updated list of Java keywords (used for escaping names in generated code)

PaulKlint committed:
Fixed typechecker warnings

jurgenvinju committed:
Merge branch 'main' into fix/issues-2691-2689-2690

jurgenvinju committed:
Merge pull request #2692 from usethesource/fix/issues-2691-2689-2690

fix/issues 2691 2689
add splitting of code blocks based on inline single line comments

PaulKlint committed:
Merge pull request #2708 from usethesource/fix/warnings2

Fixed typechecker warnings

PaulKlint committed:
Root cause of this issues lmMloc > lmTpl should be lmMloc >= lmTpl;

PaulKlint committed:
Auxiliary safety measure.

jurgenvinju committed:
removed RuntimeException rethrow around initializing a module, and removed setInitialized(false) for it. In the current setup a module is initialized with and without having detected errors. This fixes #2714, but needs to be thoroughly tested

jurgenvinju committed:
Revert "removed RuntimeException rethrow around initializing a module, and removed setInitialized(false) for it. In the current setup a module is initialized with and without having detected errors. This fixes #2714, but needs to be thoroughly tested"

This reverts commit 98c8a1d2bf87cd21863bb1219ef7c92ffb966128.

jurgenvinju committed:
Merge branch 'main' into fix/again-issue-2714

jurgenvinju committed:
fix issue #2714 again

toinehartman committed:
Fix test configs and utils (#2712)

* Do not import test configs in production.

* Consistent test config paths.

* Whitespace.

* Reuse.

* Import test configs in test utils.

* Add imports where needed.

* Add another missing import.

toinehartman committed:
Fix Maven separators in docs and messages. (#2719)

DavyLandman committed:
Fixed a bug where the root of a jar+file or zip uri was always existing, even if the underlying file did not exist (#2720)

PaulKlint committed:
Commented out println

PaulKlint committed:
Undo previous > to >= change

PaulKlint committed:
Improved computation of require definitions

PaulKlint committed:
Small refactoring and added asserts

toinehartman committed:
Failing test for #2723.

toinehartman committed:
Replace deprecated `@loc` anno with `src` kw param

Fixes #2723

jurgenvinju committed:
Merge pull request #2724 from usethesource/fix/2723-loc-annos-on-concrete-trees

Replace deprecated loc annos on concrete trees

DavyLandman committed:
Make sure files cannot race on their timestamp

DavyLandman committed:
Porting fixes over from the branch

DavyLandman committed:
Merge pull request #2725 from usethesource/fix/breaking-test-2710

Fix #2710

jurgenvinju committed:
started on new test for stored parser usage during import

jurgenvinju committed:
fixing

jurgenvinju committed:
improving the test

jurgenvinju committed:
made test fail as reported

DavyLandman committed:
Using ubicloud in rascal runners

jurgenvinju committed:
fixes issues with escapes in module package names while generating parsers

jurgenvinju committed:
fixes another issue with escapes in module package names while generating parsers

jurgenvinju committed:
Merge branch 'main' into fix/add-parser-load-test

jurgenvinju committed:
Merge pull request #2729 from usethesource/fix/add-parser-load-test

fix/add parser load test

jurgenvinju committed:
Merge branch 'main' into fix/again-issue-2714

jurgenvinju committed:
removed debug println

PaulKlint committed:
Removed references to testRoot

PaulKlint committed:
Added changeMultipleTimes test

PaulKlint committed:
- Renamed MStatus to ModuleProperty
- Removed unused properties
- Added property tpl_from_library: used to block recompilation of
  library modules
- Added utility functions for handling properties: hasProperty,
  hasAnyProperty, hasNotProperty, addProperty, deleteProperty
- Replaced handling of properties by the above functions

PaulKlint committed:
Merge branch 'main' into fix/changeMultipleTimes

PaulKlint committed:
Added typepal-0.16.6-RC1

PaulKlint committed:
Added typepal-0.16.6-RC1

PaulKlint committed:
Changed Rascal version in test

DavyLandman committed:
Fixed broken debug function due to bad cherry-picking

jurgenvinju committed:
removed unused monitor clone (also unused in generated code)

jurgenvinju committed:
Merge branch 'main' into fix/again-issue-2714

jurgenvinju committed:
moved NullRascalMonitor out of interpreter-specific context to generic IRascalMonitor package

jurgenvinju committed:
forgot to add

jurgenvinju committed:
Merge branch 'main' into fix/remove-monitor-clone

jurgenvinju committed:
Merge pull request #2733 from usethesource/fix/remove-monitor-clone

removed unused monitor clone (also unused in generated code)

jurgenvinju committed:
Merge branch 'main' into fix/again-issue-2714

jurgenvinju committed:
Merge pull request #2717 from usethesource/fix/again-issue-2714

fix issue #2714 again

jurgenvinju committed:
[maven-release-plugin] prepare release v0.42.2-RC1

jurgenvinju committed:
[maven-release-plugin] prepare for next development iteration

PaulKlint committed:
Cleanup and systematic use of hasProperty-like functions

PaulKlint committed:
Merge branch 'main' into fix/changeMultipleTimes

PaulKlint committed:
Fixed syntax error

PaulKlint committed:
Another syntax error

PaulKlint committed:
Fixed type error

PaulKlint committed:
Fixed error

PaulKlint committed:
Fix error

PaulKlint committed:
Minor clean-up of outdated code.

DavyLandman committed:
Merge pull request #2732 from usethesource/fix/changeMultipleTimes

Fix/change multiple times

DavyLandman committed:
Make sure to respect the terminal capabilities when building a REPL

jurgenvinju committed:
execute global initialization after all the other declarations have finished

jurgenvinju committed:
do not cache global functions during initialization anymore

jurgenvinju committed:
added test for current PR

jurgenvinju committed:
added test

jurgenvinju committed:
Merge pull request #2740 from usethesource/fix/global-init

execute global initialization after all the other declarations have finished

jurgenvinju committed:
Merge branch 'main' into fix/no-cache-during-loading

jurgenvinju committed:
Merge pull request #2741 from usethesource/fix/no-cache-during-loading

do not cache global functions during initialization anymore

PaulKlint committed:
Improved two calculator names

PaulKlint committed:
Add return type as dependency for check on return statement

PaulKlint committed:
Merge branch 'main' into fix/missing-dependency-check-return-statement

PaulKlint committed:
Merge pull request #2742 from usethesource/fix/missing-dependency-check-return-statement

Fix/missing dependency check return statement

PaulKlint committed:
Update to typepal-0.16.7 (#2746)

* Update to typepal-0.16.7

* Fix changed error message

DavyLandman committed:
[maven-release-plugin] prepare release v0.42.2-RC2

DavyLandman committed:
[maven-release-plugin] prepare for next development iteration

toinehartman committed:
Summary overload given TPL location

toinehartman committed:
Remove commented import.

toinehartman committed:
Refactor doc tags.

jurgenvinju committed:
Merge pull request #2748 from usethesource/fix/summary-from-loc

Add overload for summary from TPL location

jurgenvinju committed:
fixed #2747 and a number of other error scenarios like wrong module names

jurgenvinju committed:
Merge branch 'main' into fix/issue-2747

jurgenvinju committed:
made module load messages observable to util::Eval::RascalRuntime

jurgenvinju committed:
added a minimal set of load tests in the presence of errors in modules, that trigger #2747

jurgenvinju committed:
organized imports

jurgenvinju committed:
added more tests

DavyLandman committed:
Merge pull request #2749 from usethesource/fix/issue-2747

fixed #2747 and a number of other error scenarios like wrong module names

DavyLandman committed:
[maven-release-plugin] prepare release v0.42.2-RC3

DavyLandman committed:
[maven-release-plugin] prepare for next development iteration

PaulKlint committed:
Failing test

PaulKlint committed:
Make signature info available before collecting signature

PaulKlint committed:
Make typed variable pattern dependent on return type

- This is needed for proper execution order
- This is rather coarse solution. Maybe move this to cases for
   type parameters?

PaulKlint committed:
Made new dependency on returnType conditional

PaulKlint committed:
Merge pull request #2751 from usethesource/fix/missing-dependency-check-typed-var-pattern

Fix/missing dependency check typed var pattern

DavyLandman committed:
[maven-release-plugin] prepare release v0.42.2

DavyLandman committed:
[maven-release-plugin] prepare for next development iteration

jurgenvinju committed:
Merge branch 'main' into setfn
